### PR TITLE
feat(upgrade): change thiserror to snafu

### PR DESCRIPTION
**Pull request by sinhaashish**
the plugin code for upgrade was using `thiserror` , this has been changed to `snafu`.

The reason behind this is to merge upgrade-job  and upgrade crate to a single crate

### CODE REVIEW

File: k8s/upgrade/src/upgrade_resources/upgrade.rs [hunk 1]
```diff
@@ -7,15 +7,23 @@ use crate::{
         UPGRADE_JOB_IMAGE_NAME, UPGRADE_JOB_IMAGE_REPO, UPGRADE_JOB_NAME_SUFFIX,
         UPGRADE_JOB_SERVICEACCOUNT_NAME_SUFFIX,
     },
-    error::Error,
+    error::{
+        ClusterRoleBindingCreate, ClusterRoleBindingDelete, ClusterRoleCreate, ClusterRoleDelete,
```
Comment (tiagolobocastro) [inline k8s/upgrade/src/upgrade_resources/upgrade.rs:11]: wow this is huge, I suggest use it with the module, example: `error::ClusterRoleBindingCreate`

File: k8s/plugin/src/main.rs [hunk 2]
```diff
@@ -164,7 +164,7 @@ async fn execute(cli_args: CliArgs) {
                 });
 
                 if resources.dry_run {
-                    let _ = resources.dummy_apply(&cli_args.namespace).await;
+                    _ = resources.dummy_apply(&cli_args.namespace).await;
```
[the remark this answers is not in the document]
Comment (sinhaashish) [reply k8s/plugin/src/main.rs:167]: It compiles.

File: k8s/upgrade/src/upgrade_resources/upgrade.rs (hunk 1 above)
Comment (sinhaashish) [reply k8s/upgrade/src/upgrade_resources/upgrade.rs:11]: done

File: k8s/plugin/src/main.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (tiagolobocastro) [reply k8s/plugin/src/main.rs:167]: awesome I didn't know :)

**Review by niladrih — APPROVED** [review]

**Comment by sinhaashish** [comment]
bors merge

**Pull request merged**